sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-factory.ts

```diff
@@ -0,0 +1,92 @@
+import { MSFAANumber } from "@sims/sims-db";
+import {
+  E2EDataSources,
+  createFakeMSFAANumber,
+  createFakeStudent,
+  createFakeUser,
+  saveFakeApplication,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import { MSFAATestInputData } from "./msfaa-part-time-process-integration.scheduler.models";
+import { createFakeSINValidation } from "@sims/test-utils/factories/sin-validation";
+
+/**
+ * Save an MSFAA record providing all data that will be
+ * part of the MSFAA file generation.
+ * @param db data source helper.
+ * @param msfaa test input data.
+ * @returns a saved MSFAA record that used th input
+ * test data to be created.
+ */
+export async function saveMSFAATestInputData(
+  db: E2EDataSources,
+  msfaaDataInput: MSFAATestInputData,
+): Promise<MSFAANumber> {
+  // User.
+  const fakeUser = createFakeUser();
+  fakeUser.firstName = msfaaDataInput.givenName;
+  fakeUser.lastName = msfaaDataInput.surname;
+  fakeUser.email = msfaaDataInput.email;
+  // Student.
+  const fakeStudent = createFakeStudent(fakeUser);
+  fakeStudent.birthDate = msfaaDataInput.birthDate;
+  fakeStudent.gender = msfaaDataInput.gender;
+  fakeStudent.contactInfo = {
+    address: {
+      addressLine1: msfaaDataInput.addressLine1,
+      addressLine2: msfaaDataInput.addressLine2,
+      city: msfaaDataInput.city,
+      country: msfaaDataInput.country,
+      selectedCountry: msfaaDataInput.country,
+      provinceState: msfaaDataInput.provinceState,
+      postalCode: msfaaDataInput.postalCode,
+    },
+    phone: msfaaDataInput.phone,
+  };
+  // SIN validation.
+  const sinValidation = createFakeSINValidation({ student: fakeStudent });
+  sinValidation.sin = msfaaDataInput.sin;
+  // Student.
+  const student = await saveFakeStudent(db.dataSource, {
+    student: fakeStudent,
+    sinValidation,
+  });
+  // Application, offering, location.
+  const referenceApplication = await saveFakeApplication(db.dataSource);
+  referenceApplication.relationshipStatus = msfaaDataInput.maritalStatus;
+  await db.application.save(referenceApplication);
+  const offering = referenceApplication.currentAssessment.offering;
+  offering.offeringIntensity = msfaaDataInput.offeringIntensity;
+  await db.educationProgramOffering.save(offering);
+  offering.institutionLocation.institutionCode = msfaaDataInput.institutionCode;
+  await db.institutionLocation.save(offering.institutionLocation);
+  // MSFAA.
+  const newMSFAANumber = createFakeMSFAANumber({
+    student,
+    referenceApplication,
+  });
+  newMSFAANumber.msfaaNumber = msfaaDataInput.msfaaNumber;
+  newMSFAANumber.dateRequested = null;
+  newMSFAANumber.dateSigned = null;
+  newMSFAANumber.serviceProviderReceivedDate = null;
+  newMSFAANumber.cancelledDate = null;
+  newMSFAANumber.newIssuingProvince = null;
```

Comment:
IMO, I feel like this logic can be kept inside the factory `createFakeMSFAANumber` based on an optional parameter provided to the factory.

Reply:
It follows the idea of having the factory create the record and adjust the test as needed.
Keep in mind that these "nulls" are also set once inside a "factory".
Is the idea to create options like "pending", "canceled", or "signed" states?